src/visualizers/widgets/TextEditor/TextEditorWidget.js

```diff
@@ -1,79 +1,111 @@
-/*globals $, define*/
+/*globals $, define, monaco */
 /*jshint browser: true*/
 
 define([
-    'ace/ace',
     'underscore',
-    './completer',
     'js/Utils/ComponentSettings',
+    'text!./MonacoLanguages.json',
+    'vs/editor/editor.main',
     'jquery-contextMenu',
     'css!./styles/TextEditorWidget.css'
 ], function (
-    ace,
     _,
-    Completer,
-    ComponentSettings
+    ComponentSettings,
+    MonacoLanguages
 ) {
     'use strict';
 
-    var TextEditorWidget,
-        WIDGET_CLASS = 'text-editor',
+    const WIDGET_CLASS = 'text-editor',
         LINE_COMMENT = {
             python: '#',
             lua: '--',
             yaml: '#'
         };
 
-    TextEditorWidget = function (logger, container, config={}) {
-        this.logger = logger.fork('Widget');
+    MonacoLanguages = JSON.parse(MonacoLanguages);
+    const DEFAULT_THEMES = ['vs-dark', 'vs', 'hc-black'];
+    const DEFAULT_COLORS = {
+        'vs': '#fffffe',
+        'vs-dark': '#1e1e1e',
+        'hc-black': '#000000'
+    };
 
+    const AVAILABLE_KEYBINDINGS = ['default'];
+
+    const TextEditorWidget = function (logger, container, config={}) {
+        this.logger = logger.fork('Widget');
+        this._registerMonacoLanguages();
         this.language = this.language || config.language || 'python';
+        this.monacoURI = this._getMonacoURI();
+        const value = config.value || "def dummy_python_func():\n\tpass";
```

Comment:
Why not just use empty string as the default?

Reply:
Okay. It should probably default to empty string.